storage/mysql/tree_storage.go

```diff
@@ -14,23 +14,23 @@ import (
 )
 
 // These statements are fixed
-const insertNodeSql string = `INSERT INTO Node(TreeId, NodeId, NodeHash, NodeRevision) VALUES (?, ?, ?, ?)`
+const insertSubtreeSql string = `INSERT INTO Subtree(TreeId, SubtreeId, Nodes, SubtreeRevision) VALUES (?, ?, ?, ?)`
 const insertTreeHeadSql string = `INSERT INTO TreeHead(TreeId,TreeHeadTimestamp,TreeSize,RootHash,TreeRevision,RootSignature)
 		 VALUES(?,?,?,?,?,?)`
 const selectTreeRevisionAtSizeSql string = "SELECT TreeRevision FROM TreeHead WHERE TreeId=? AND TreeSize=? ORDER BY TreeRevision DESC LIMIT 1"
 const selectActiveLogsSql string = "select TreeId, KeyId from Trees where TreeType='LOG'"
 const selectActiveLogsWithUnsequencedSql string = "SELECT DISTINCT t.TreeId, t.KeyId from Trees t INNER JOIN Unsequenced u WHERE TreeType='LOG' AND t.TreeId=u.TreeId"
 
-const selectNodesSql string = `SELECT x.NodeId, x.MaxRevision, Node.NodeHash
-				 FROM (SELECT n.NodeId, max(n.NodeRevision) AS MaxRevision
-							 FROM Node n
-							 WHERE n.NodeId IN (` + placeholderSql + `) AND
+const selectSubtreeSql string = `SELECT x.SubtreeId, x.MaxRevision, Subtree.Nodes
```

Comment:
I think this query will perform similarly to the get nodes one. Don't worry about it now, we know we need to do optimization.